src/java/org/apache/cassandra/cql3/statements/ModificationStatement.java

```diff
@@ -854,18 +854,23 @@ private UpdateParameters makeUpdateParameters(Collection<ByteBuffer> keys,
     }
 
     private Slices toSlices(SortedSet<ClusteringBound<?>> startBounds, SortedSet<ClusteringBound<?>> endBounds)
+    {
+        return toSlices(metadata, startBounds, endBounds);
+    }
+
+    public static Slices toSlices(TableMetadata metadata, SortedSet<ClusteringBound<?>> startBounds, SortedSet<ClusteringBound<?>> endBounds)
     {
         assert startBounds.size() == endBounds.size();
 
-        Slices.Builder builder = new Slices.Builder(metadata().comparator);
+        Slices.Builder builder = new Slices.Builder(metadata.comparator);
```

Comment:
nit: I know this is fine because of local precedence, but perhaps we can remove the `metadata` ambiguity w/ some creative naming :)